specification/_types/mapping/types/complex/object/ObjectProperty.ts

```diff
@@ -28,5 +28,5 @@ export class ObjectProperty extends CorePropertyBase {
   dynamic?: boolean | DynamicMapping
   enabled?: boolean
   properties?: Dictionary<PropertyName, Property>
-  type: 'object'
+  type?: 'object'
```

Comment:
This is the right `object`. Re adding previous comment:

> As discussed with @delvedor, we only need this single line. This is because field mapping uses object as the default type (see bottom of this page). In all other cases (other variants and containers that don't have a default type) type should be require.